elastic/kibana
# [Maps] deprecate 'map.regionmap' kibana config and 'Configured GeoJSON' source

Part of https://github.com/elastic/kibana/issues/97532

This PR:
1) Registers deprecation messages for 'map.regionmap' kibana config with upgrade assistant.
2) Removes "Configured GeoJSON" layer wizard
3) Updates "Configured GeoJSON" layers to show "alert" icon with tooltip explaining how to resolve alert.

<img width="400" alt="Screen Shot 2021-06-25 at 6 14 00 AM" src="https://user-images.githubusercontent.com/373691/123423272-93f68600-d57c-11eb-916f-9251429e1416.png">

<img width="600" alt="Screen Shot 2021-06-24 at 4 40 43 PM" src="https://user-images.githubusercontent.com/373691/123342330-c8caf480-d50c-11eb-9eac-014bc2f83830.png">

To test:
1) add the following to your kibana.dev.yml
```
xpack.upgrade_assistant.readonly: false

map.regionmap:
 layers:
  - attribution: "EMS"
    name: "My copy of world countries"
    url: "https://vector.maps.elastic.co/files/world_countries_v7.geo.json?elastic_tile_service_tos=agree&my_app_name=ems-landing-page&my_app_version=7.10.0&license=643c1faf-80fc-4ab0-9323-4d9bd11f4bbc"
    fields:
     - name: "name"
       description: "Name"
     - name: "iso2"
       description: "iso2"
```
2) Import map saved object with configured geojson layer
```
{"attributes":{"description":"","layerListJSON":"[{\"sourceDescriptor\":{\"type\":\"EMS_TMS\",\"isAutoSelect\":true},\"id\":\"85be2442-49e4-4a66-a6d0-81c0fc22c26a\",\"label\":null,\"minZoom\":0,\"maxZoom\":24,\"alpha\":1,\"visible\":true,\"style\":{\"type\":\"TILE\"},\"includeInFitToBounds\":true,\"type\":\"VECTOR_TILE\"},{\"sourceDescriptor\":{\"type\":\"REGIONMAP_FILE\",\"name\":\"My copy of world countries\"},\"id\":\"5d766667-52b9-4984-a33e-38d19308498d\",\"label\":null,\"minZoom\":0,\"maxZoom\":24,\"alpha\":0.75,\"visible\":true,\"style\":{\"type\":\"VECTOR\",\"properties\":{\"icon\":{\"type\":\"STATIC\",\"options\":{\"value\":\"marker\"}},\"fillColor\":{\"type\":\"STATIC\",\"options\":{\"color\":\"#54B399\"}},\"lineColor\":{\"type\":\"STATIC\",\"options\":{\"color\":\"#41937c\"}},\"lineWidth\":{\"type\":\"STATIC\",\"options\":{\"size\":1}},\"iconSize\":{\"type\":\"STATIC\",\"options\":{\"size\":6}},\"iconOrientation\":{\"type\":\"STATIC\",\"options\":{\"orientation\":0}},\"labelText\":{\"type\":\"STATIC\",\"options\":{\"value\":\"\"}},\"labelColor\":{\"type\":\"STATIC\",\"options\":{\"color\":\"#000000\"}},\"labelSize\":{\"type\":\"STATIC\",\"options\":{\"size\":14}},\"labelBorderColor\":{\"type\":\"STATIC\",\"options\":{\"color\":\"#FFFFFF\"}},\"symbolizeAs\":{\"options\":{\"value\":\"circle\"}},\"labelBorderSize\":{\"options\":{\"size\":\"SMALL\"}}},\"isTimeAware\":true},\"includeInFitToBounds\":true,\"type\":\"VECTOR\",\"joins\":[]}]","mapStateJSON":"{\"zoom\":2.01,\"center\":{\"lon\":0,\"lat\":19.94277},\"timeFilters\":{\"from\":\"now-15m\",\"to\":\"now\"},\"refreshConfig\":{\"isPaused\":true,\"interval\":0},\"query\":{\"query\":\"\",\"language\":\"kuery\"},\"filters\":[],\"settings\":{\"autoFitToDataBounds\":false,\"backgroundColor\":\"#ffffff\",\"disableInteractive\":false,\"disableTooltipControl\":false,\"hideToolbarOverlay\":false,\"hideLayerControl\":false,\"hideViewControl\":false,\"initialLocation\":\"LAST_SAVED_LOCATION\",\"fixedLocation\":{\"lat\":0,\"lon\":0,\"zoom\":2},\"browserLocation\":{\"zoom\":2},\"maxZoom\":24,\"minZoom\":0,\"showScaleControl\":false,\"showSpatialFilters\":true,\"showTimesliderToggleButton\":true,\"spatialFiltersAlpa\":0.3,\"spatialFiltersFillColor\":\"#DA8B45\",\"spatialFiltersLineColor\":\"#DA8B45\"}}","title":"map with configured geojson layer","uiStateJSON":"{\"isLayerTOCOpen\":true,\"openTOCDetails\":[]}"},"coreMigrationVersion":"8.0.0","id":"20111bb0-d5ae-11eb-a404-b351e49a7649","migrationVersion":{"map":"7.14.0"},"references":[],"type":"map","updated_at":"2021-06-25T12:09:10.782Z","version":"WzI0LDFd"}
{"excludedObjects":[],"excludedObjectsCount":0,"exportedCount":1,"missingRefCount":0,"missingReferences":[]}
```
3) Open imported map saved object. Notice the layer no longer displays data and has an alert icon in the TOC.
4) Open upgrade assistant. Notice there is a critical warning for removing `map.regionmap`
